ion/unmarshal.go

```diff
@@ -671,3 +701,48 @@ func indirect(v reflect.Value, wantPtr bool) reflect.Value {
 
 	return v
 }
+
+func (d *Decoder) decodeToStructWithAnnotation(v reflect.Value) error {
+	if err := d.attachAnnotations(v); err != nil {
+		return err
+	}
+
+	fields := fieldsFor(v.Type())
+	for _, field := range fields {
+		if field.name != AnnotationsTag {
+			field := findField(fields, field.name)
+			if field != nil {
+				subValue, err := findSubvalue(v, field)
+				if err != nil {
+					return err
+				}
+
+				if err := d.decodeTo(subValue); err != nil {
+					return err
+				}
+				break
+			}
+		}
+	}
+	return nil
+}
+
+func (d *Decoder) attachAnnotations(v reflect.Value) error {
+	fields := fieldsFor(v.Type())
+	field := findField(fields, AnnotationsTag)
```

Comment:
I think we want to look for a field with an "annotations" opt (in the style of `omitempty`), rather than relying on the field's name? A struct might legitimately have a field named "annotations" which would lead to surprising behavior.

```go
type LegalDocument struct {
  Text string
  Annotations string  // oops this will explode when we try to set it to a []string
}
```

Reply:
That is right; however, I am not sure if I understood your suggestion correctly.
Are you suggesting to change ```annotations``` to ```,annotations```? Or adding another field to the expected struct?

Reply:
Correct, users would do `ion:",annotations"` (the actual field name doesn't matter since the struct's not actually getting serialized). The `field` struct would get an extra bool field named `annotations` that would get filled in like `omitempty` does; this code would look for a field with that opt set to true.

Reply:
Updated the PR.
Thanks.